### PR TITLE
Explanations of why we introduce topics

### PR DESCRIPTION
**Pull request by cbrantley4**
I've gotten started on this ticket. Let me know what you think

### CODE REVIEW

File: Section03/3.1-Functions.md [hunk 1]
```diff
@@ -5,11 +5,23 @@
...
+You can right-click on any website in Google Chrome to inspect the page. Once you select the console tab, you will see a clear page. This is a blank slate where you can type JavaScript.
+</br>
 - JS Function Declaration, Invocation, and Return
 - JS Parameters and Arguments
+<br>
+It is possible to name the input variables in a function, known as parameters. The function is then run with values in place of these variables. These are the arguments of the function.
+</br>
 - Scope in JS functions
+<br>
+When a function is created it is necessary to define what the expected output is. This expected output is code that is only able to be used inside the function.
```
Comment (JohanBester) [inline Section03/3.1-Functions.md:18]: Hi Cameron,
In the old curriculum we had this phrase about scope...
" In JavaScript, scopes are created by code blocks and modules, like in a function. A variable defined inside a scope is accessible only within that scope, but inaccessible outside of it. "
Not sure if something like this is simpler in the context?

I was thinking that the current description could be confusing as the output that is returned from a function can be used elsewhere.

File: Section03/3.2-DataCollections.md [hunk 2]
```diff
@@ -5,8 +5,18 @@
 In this module we will look at ...
 
 - Primitive data types and collections of data
+<br>
+Primitive Data Types refers to all pieces of data you will see in the JavaScript environment. You can create a collection of these pieces of data by grouping them as objects.
```
Comment (JohanBester) [inline Section03/3.2-DataCollections.md:9]: Don't want to be picky, but not all pieces of data one will see in JS are primitives.

**Review by JohanBester — COMMENTED** [review]
Hello Cameron,

I have looked at the changes you are proposing. You have done a lot of work here. Thank you for your dedication..

I was wondering if it is totally necessary to comment on each Bullitt point in every section though?

I though the intent of the ticket was to just give a simple, 1 paragraph, reason for why we are learning the materials in a certain section. It was certainly not meant to write a reason for every bullet point... 
EG: Why are we learning about data data collections? Or why are we learning about Array methods? Or why are we learning about HTML or CSS? Or why do we have to Learn about SPAs?

I think by narrowing down the amount and number of comments,(1) it will be way less work, and (2) the comments can be more targeted in that they will cover a broader scope with less detailed. 
The detail belongs in the lesson, not the into paragraph that was meant to only outline what the class is about.
Else the opening outline is redundant, and more detail should be added into the lesson.

I feel we are trying to reinvent the wheel here, and "muddying the waters" with too much info before we even get into teaching the details of what the students should learn.

My opinions are aimed at reducing the work you are doing, and making the curiculum more maintainable.

File: Section03/3.2-DataCollections.md (hunk 2 above)
Comment (cbrantley4) [reply Section03/3.2-DataCollections.md:9]: I'm glad you're being picky!!

File: Section03/3.1-Functions.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ In this module we'll look at ...
 - 'Pure' Functions and functional programing
 - Callback Functions and (fat) Arrow Syntax
 
+In JavaScript, scopes are created by code blocks and modules, like in a function. A variable defined inside a scope is accessible only within that scope, but inaccessible outside of it.
```
Comment (matsinet) [inline Section03/3.1-Functions.md:14]: We need to update this to state the variable in parent scopes are available in child scopes.    It is accurate that variables are not available in child or sibling scopes.

File: Section03/3.2-DataCollections.md [hunk 4]
```diff
@@ -11,6 +11,9 @@ In this module we will look at ...
 - Mutable vs immutable data elements
 - The Reference vs. the Value of a data element
 - Useful String methods
+<br>
+
+JS Arrays and Objects are both collections of data. Each individual piece of data inside a JS Array can be identified by the index. Each individual piece of data inside a JS Object, however, is identified by its key value as the data is written in key-value pairs.
```
Comment (matsinet) [inline Section03/3.2-DataCollections.md:16]: I would remove the word value when stating the an object data is identified such as:
"identified by its key as the data"

**Review by matsinet — CHANGES_REQUESTED** [review]

**Review by matsinet — APPROVED** [review]
👍

**Review by dadler89 — APPROVED** [review]

**Pull request merged**